cmd/controller/main.go

```diff
@@ -93,7 +93,7 @@ func main() {
 	// Build all of our controllers, with the clients constructed above.
 	// Add new controllers to this array.
 	// You also need to modify numControllers above to match this.
-	controllers := []*kncontroller.Impl{
+	controllersArray := [...]*kncontroller.Impl{
```

Comment:
This might be clearer as `[numControllers]*kncontroller.Impl`. That will also catch the issue of `numControllers < len(controllersArray)`.